internal/pkg/cli/task_run.go

```diff
@@ -501,6 +516,30 @@ func (o *runTaskOpts) confirmSecretsAccess() error {
 	return nil
 }
 
+func (o *runTaskOpts) validateEnvCompatibilityForGenerateJobCmd(app, env string) error {
+	envStack, err := o.envCompatibilityChecker(app, env)
+	if err != nil {
+		return err
+	}
+	version, err := envStack.Version()
+	if err != nil {
+		return fmt.Errorf("retrieve version of environment stack %q in application %q: %v", env, app, err)
+	}
+	// The '--generate-cmd' flag was introduced in env v1.4.0. In env v1.8.0,
+	// EnvManagerRole took over, but "states:DescribeStateMachine" permissions
+	// weren't added until 1.12.2. Any version predating the feature's
+	// introduction or between v1.8.0 and 1.12.2 are invalid.
+	if semver.Compare(version, "v1.12.2") < 0 {
```

Comment:
Maybe make "v1.12.2" a constant and move the comment there as well?

Reply:
I will simplify the comment here instead.